lib/SPIRVProducerPass.cpp

```diff
@@ -1613,7 +1609,7 @@ void SPIRVProducerPass::FindConstantPerFunc(Function &F) {
         }
 
         continue;
-      } else if ((Instruction::Xor == I.getOpcode()) && I.getType()->isIntegerTy(1)) {
+      } else if ((Instruction::Xor == I.getOpcode()) && I.getType()->isIntOrIntVectorTy(1)) {
```

Comment:
Within this if we iterate through the operands of `I`. If any operand is a constant it is a assumed to be a ConstantInt. That `cast` seems like it would be invalid in that case.

Reply:
I've reverted this change. This handling of constants only exists to accompany the special case for `Instruction::Xor` in `SPIRVProducerPass::GenerateInstruction`. Since that can't meaningfully be updated to deal with vectors as well, the corresponding handling of constants shouldn't be changed.